language/vm/vm_runtime/vm_runtime_types/src/value.rs

```diff
@@ -524,7 +600,9 @@ impl Struct {
 
     fn equals(&self, s2: &Struct) -> VMResult<bool> {
         if self.0.len() != s2.0.len() {
-            return Err(VMStatus::new(StatusCode::INTERNAL_TYPE_ERROR));
+            // REVIEW: should this return false?
```

Comment:
Error makes sense to me as this should be rejected by the verifier at the beginning. cc @sblackshear

Reply:
removed the comment